.github/workflows/deploy-core.yaml

```
@@ -0,0 +1,208 @@
+        run: |
+          pnpm vercel --token=${{ secrets.VERCEL_TOKEN }} env pull --environment=preview --git-branch=${{ env.VERCEL_BRANCH }} .env
+
+      - name: Run Tests
+        run: pnpm ci:test
+
+  test-prod:
+    if: ${{ github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)